### DISCUSSION TITLE
How to import resources by Name in provider-aws

### DISCUSSION THREAD

**Discussion by ytsarev** [crossplane/crossplane#2229] [Q&A] [answered]
Hey guys, i’m importing some core existing resources like vpc and subnets following https://crossplane.io/docs/v1.1/concepts/managed-resources.html#importing-existing-resources  to further reference them in the Compositions.

I’m using `provider-aws`.

It works but only with resource Ids.

Is there a way to import resources by a Name tag ? The main motivation is to avoid hard-coded Ids in the manifests

**Comment by muvaf** [comment, marked as answer]
There isn't such a feature currently. The value of external-name annotation is the name of the resource if it can be named but there are many resources like VPC which cannot be named by the user.

**Reply by ytsarev** [reply]
thanks a lot for clarification
